docs/src/man/comparisons.md

```diff
@@ -235,6 +235,59 @@ The table below compares more advanced commands:
+|Rename multiple columns (in place) |`setnames(df, c("x", "y"), c("x_new", "y_new")`|`rename!(df, [:x, :y] => [:x_new, :y_new])`|
+|Pick columns            |`df[, list(x, y)]`          |`select(df, :x, :y)`|
+|Remove columns          |`df[, -"x" ]`               |`select(df, Not(:x))`|
+|Remove columns (in place) |`df[, c("x") := NULL ]`   |`select!(df, Not(:x))`|
+|Pick & transform columns|`df[, list(mean(x), y)]`    |`select(df, :x => mean, :y)`|
+|Pick rows               |`df[ x >= 1 ]`              |`filter(:x => >=(1), df)`|
+|Sort rows (in place)    |`setorder(df, x)`           |`sort!(df, :x)`|
+|Sort rows               |`df[ order(x) ]`            |`sort(df, :x)`|
+
+### Grouping data and aggregation
+
+|Operations              | data.table                                      | DataFrames.jl     |
+|:-----------------------|:------------------------------------------------|:------------------|
+|Reduce multiple values  |`df[, list(mean(x)), by = list(id) ]`            |`combine(groupby(df, :id), :x => mean)`|
+|Add new columns (in place) |`df[, x_mean := mean(x), by = list(id) ]`        |`transform!(groupby(df, :id), :x => mean)`|
+|Pick & transform columns|`df[, list(x_mean = mean(x), y), by = list(id) ]`|`select(groupby(df, :id), :x => mean, :y)`|
+
+### More advanced commands
+
+|Operations              | data.table                                      | DataFrames.jl             |
+|:-----------------------|:------------------------------------------------|:--------------------------|
+|Complex Function          |`df[, list(mean(x, na.rm = T)) ]`              |`combine(df, :x => x -> mean(skipmissing(x)))`|
+|Transform certain rows (in place) |`df[x<=0, x:=0 ]`                      |`transform!(df, :x => (x -> ifelse.(x .<= 0, 0, x)) => :x)`|
+|Transform several columns |`df[, list(max(x), min(y)) ]`                  |`combine(df, :x => maximum,  :y => minimum)`|
+|                          |`df[, lapply(.SD, mean), .SDcols = c("x", "y") ]`   |`combine(df, [:x, :y] .=> mean)`|
+|                          |`df[, lapply(.SD, mean), .SDcols = patterns("x*") ]`|`combine(df, names(df, r"^x") .=> mean)`|
+|                          |`df[, unlist(lapply(.SD, function(x) c(max=max(x), min=min(x)))), .SDcols = c("x", "y") ]`|`combine(df, ([:x, :y] .=> [maximum minimum])...)`|
+|Multivariate function     |`df[, list(cor(x,y)) ]`                        |`transform(df, [:x, :y] => cor)`|
+|Row-wise                  |`df[, min_xy := min(x, y), by = 1:nrow(df)]`   |`transform(df, [:x, :y] => ByRow(min))`|
+|                          |`df[, argmax_xy := which.max(.SD) , .SDcols = patterns("x*"), by = 1:nrow(df) ]`|`transform(df, AsTable(r"^x") => ByRow(argmax))`|
+|DataFrame as input        |`df[, head(.SD, 2) ]`                          |`combine(d -> first(d, 2), df)`|
+|DataFrame as output       |`df[, .SD[, .(value = min(x), max(x))] ]`      |`combine(:x => x -> (value = [minimum(x), maximum(x)]), df)`|
```

Comment:
same here - there is something wrong with this example (for sure DataFrames.jl example is not working)

Reply:
Same as above. I think this is useful when doing grouped operation. 
When I use DataFrames though I prefer to iterate over the subgroups and run my operation in a loop for clarity (at the cost of conciseness). So perhaps this is not really the place for such example and a separate long and detailed post could go into such details.

Reply:
I meant that:
```
combine(:x => x -> (value = [minimum(x), maximum(x)]), df)
```
is not a valid syntax any more. You probably meant to write (note the comma after `]` otherwise `value = ` is just an assignment to a local variable that will get ignored):
```
julia> combine(df, :x => (x -> (value = [minimum(x), maximum(x)],)) => AsTable)
```
or
```
julia> combine(df, :x => (x -> [minimum(x), maximum(x)]) => :value)
```
btw. this specific case can be also written as:
```
julia> combine(df, :x => collect∘extrema => :value)
```